bokeh/document/locking.py

```diff
@@ -53,7 +53,7 @@
 
 F = TypeVar("F", bound=Callable[..., Any])
 
-class NoLockCallback(Protocol[F]): # type: ignore # F_co, needs F & {nolock: true}
```

Comment:
mypy said this `ignore` was unused, and everything seems to work with it removed. Perhaps an old mypy limitation was lifted?

Reply:
If it was, it needs to be reflected in `mypy >= x.y.z` in dependencies, otherwise it's easy to leave older version of mypy around and get spurious errors.

Reply:
@mattpap can you elaborate on the previous comment? Was there an explicit error emitted when running `mypy`? If not how was the need evidenced?

Reply:
I'm don't remember now the specifics. I would just bump minimum required mypy to the latest on conda-forge (0.910).